src/main/java/org/terasology/simpleFarming/systems/TreeAuthoritySystem.java

```diff
@@ -0,0 +1,573 @@
+		}
+		return leafEntity;
+	}
+	
+	/**
+	 * Destroys all of the blocks in a tree except for the root. This is done usually to make way for a new growth stage of the tree.
+	 * 
+	 * @param rootEntity The root of the tree.
+	 * @param location The root's location.
+	 */
+	private void destroyTree(EntityRef rootEntity, Vector3i location) {
+		location.addY(1);
+		EntityRef aboveLogEntity = blockEntityRegistry.getExistingEntityAt(location);
+		location.addY(-1);
+		if(aboveLogEntity != EntityRef.NULL && aboveLogEntity.hasComponent(LogComponent.class)
+				&& aboveLogEntity.getComponent(LogComponent.class).root == rootEntity) {
+			destroyLog(null, aboveLogEntity);
+		}
+	}
+
+	/**
+	 * Destroys a log block in a tree, and all of the logs above it along with all of the leaves on a tree. Also marks the tree as
+	 * no longer alive so that it does not grow anymore.
+	 * 
+	 * @param event The event for the log being destroyed. If it is present, it will be sent to all other blocks which are destroyed,
+	 * allowing them to drop items, otherwise, the blocks will simply be replaced with air, allowing for growth or un-growth.
+	 * @param log The log to be destroyed.
+	 * 
+	 * @see #destroyLeaves(Set, DoDestroyEvent)
+	 */
+	private void destroyLog(DoDestroyEvent event, EntityRef log) {
+		LogComponent logComponent = log.getComponent(LogComponent.class);
+		EntityRef rootEntity = logComponent.root;
+		RootComponent rootComponent = rootEntity.getComponent(RootComponent.class);
+		
+		if(rootComponent.alive) {
+			destroyLeaves(rootComponent.leaves, event);
+			rootComponent.leaves.clear();
+			rootComponent.alive = false;
+			rootEntity.addOrSaveComponent(rootComponent);
+		}
+		
+		if(event == null) {
+			worldProvider.setBlock(logComponent.location, airBlock);
+		}
+		
+		Vector3i aboveLocation = new Vector3i(logComponent.location).addY(1);
+		EntityRef aboveLogEntity = blockEntityRegistry.getExistingEntityAt(aboveLocation);
+		if(aboveLogEntity != EntityRef.NULL && aboveLogEntity.hasComponent(LogComponent.class)
+				&& aboveLogEntity.getComponent(LogComponent.class).root == rootEntity) {
+			if(event == null) {
+				destroyLog(null, aboveLogEntity);
+			} else {
+				aboveLogEntity.send(event);
+			}
+		}
+	}
+
+	/**
+	 * Destroys all of the leaves in a set of leaves.
+	 * 
+	 * @param leaves The set of leaves to destroy.
+	 * @param event The event for the leaves being destroyed. If it is present, it will be sent to all blocks which are destroyed,
+	 * allowing them to drop items, otherwise, the blocks will simply be replaced with air, allowing for growth or un-growth.
+	 * 
+	 * @see RootComponent#leaves
+	 */
+	private void destroyLeaves(Set<EntityRef> leaves, DoDestroyEvent event) {
+		for(EntityRef leaf : leaves) {
+			if(leaf.exists() && leaf.hasComponent(BushDefinitionComponent.class)) {
+				Vector3i leafLocation = leaf.getComponent(BlockComponent.class).position;
+				if(event != null) {
+					leaf.send(event);
```

Comment:
In particular there is a method for destroying a plant, `DoDestroyPlant` that should be used for this instead of `DoDestroyEvent`